plugins/GSdx/GSTextureCache.cpp

```diff
@@ -867,7 +849,7 @@ void GSTextureCache::InvalidateVideoMem(GSOffset* off, const GSVector4i& rect, b
 				}
 				else
 				{
-					m_dst[type].erase(j);
+					i = list.erase(i);
```

Comment:
As for this loop I agree with you, the code here is branch-wise so complex that the standard pattern is not suited at all. The "continue" comment was because I already noticed that and I thought that refactoring the code a bit would help. Easiest solution though is to use old erase pattern here.